platforms/web/src/containers/AppRoutes/AppRoutes.tsx

```diff
@@ -54,7 +53,8 @@ export default function AppRoutes() {
   useNotifications();
 
   if (userData.user && !userData.loading && window.location.href.includes('#token')) {
-    return <Navigate to={`${PATH_HOME}${location.search}`} />; // component instead of hook to prevent extra re-renders
+    // leave the location intact and only remove the token from the hash
+    return <Navigate to={{ ...location, hash: '' }} replace />; // component instead of hook to prevent extra re-renders
```

Comment:
1. You can use the router's location instead of `window.location`
2. No need to pass in the previous location, only set want you want to change
3. Remove the entry from the history by adding `replace` 

```suggestion
  if (userData.user && !userData.loading && location.hash.includes('#token')) {
    // leave the location intact and only remove the token from the hash
    return <Navigate to={{ hash: '' }} replace />; // component instead of hook to prevent extra re-renders
  }
```

Reply:
1. It is router's `location`. https://github.com/jwplayer/ott-web-app/pull/499/files/ff89ff6e43f86fda891ca0e89e5b3264852aca62#diff-f5da7eac53f6186c62433bfc665c384dac9bd3ff37eac17a58260b0fa874f806R40
2. I did that at first, but unfortunately I have to include the search params explicitly, otherwise they get excluded. So I have to either go with `{ ...location,` or with `, search: location.search }`. 
3. `replace` is already there, look again. :)

Reply:
1. I mean the condition: `window.location.href.includes('#token')`
2. Hmm, that's unexpected... You can leave it as is 😄 
3. Whoops, you're right!